sdks/java/extensions/sql/jdbc/src/test/java/org/apache/beam/sdk/extensions/sql/jdbc/BeamSqlLineIT.java

```diff
@@ -0,0 +1,328 @@
+        pool.submit(
+            (Callable)
+                () -> {
+                  String[] args =
+                      buildArgs(
+                          String.format(createPubsubTableStatement, publicTopic),
+                          setProject,
+                          "SELECT COUNT(*) AS num_events,\n"
+                              + "              SUM(taxi_rides.payload.meter_increment) as revenue,\n"
+                              + "              HOP_END(\n"
+                              + "                 taxi_rides.event_timestamp, \n"
+                              + "                 INTERVAL '1' SECOND, \n"
+                              + "                 INTERVAL '2' SECOND) as minute_end\n"
+                              + "              FROM taxi_rides\n"
+                              + "              GROUP BY HOP(\n"
+                              + "                 taxi_rides.event_timestamp,\n"
+                              + "                 INTERVAL '1' SECOND, \n"
+                              + "                 INTERVAL '2' SECOND) LIMIT 2");
+                  ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                  BeamSqlLine.runSqlLine(args, null, outputStream, null);
+                  return toLines(outputStream);
+                });
+
+    // LIMIT 2 is supposed to be parsed into 6 lines.
+    assertTrue(expectedResult.get().size() == 6);
+  }
+
+  /**
+   * Check if topic exists {@param attempts} times. There is 1 sec wait time between two attempts.
+   *
+   * @param attempts number of attempts. Has to be a value > 0
+   */
+  private void checkIfTopicExists(TestPubsub eventsTopic, int attempts)
+      throws InterruptedException, IllegalArgumentException, IOException {
+    if (attempts <= 0) {
+      throw new IllegalArgumentException(
+          String.format("Set %d attempts, which should be > 0", attempts));
+    }
+
+    while (attempts > 0) {
+      List<SubscriptionPath> topics =
+          eventsTopic.listSubscriptions(
+              projectPathFromPath(String.format("projects/%s", project)), eventsTopic.topicPath());
+      if (topics.size() > 0) {
+        return;
+      }
+
+      attempts--;
+      // Sleep 1 sec
+      Thread.sleep(1000);
+    }
+
+    throw new RuntimeException(
+        String.format("Failed to create subscription within %d attempts", attempts));
+  }
+
+  private long convertTimestampToMillis(String timestamp) throws ParseException {
+    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+    return dateFormat.parse(timestamp).getTime();
+  }
+
+  private PubsubMessage message(long timestampInMillis, String jsonPayload) {
+    return new PubsubMessage(
+        jsonPayload.getBytes(UTF_8), ImmutableMap.of("ts", String.valueOf(timestampInMillis)));
+  }
+
+  private static class PubsubMessageJSONStringConstructor {
```

Comment:
nit: We will probably need a class like this at some point, but looking at this test I only see it used with one schema. I think it would be easier to understand if it was a hardcoded schema wrapped in a function, this way I don't need to understand how `PubsubMessageJSONStringConstructor` works or look up what fields it has in this instance, e.g.:

```
public String taxiRideJson(Strring rideId, double lattitude, ... etc ...) {
   JSONObject jsonObject = new JSONObject();
   jsonObject.put("ride_id", rideId);
   jsonObject.put("lattitude", lattitude);
   ...
   return jsonObject.toString();
}
```

Reply:
Or may be even create a data object for it:

```
static class TaxiRide implements Serializable {
    @JsonProperty("ride_id") String rideId;
    double lattitude;
    ...
}
```
(maybe even with `@AutoValue`).

And then just set the values and use `ObejctMapper` to convert to string

Reply:
I will go with hard coded schema wrapped in a function approach. Will also replace `fastjson` with `Jackson`.